src/Orckestra.Composer/Services/ComposerContext.cs

```diff
@@ -243,9 +253,35 @@ private void InitializeCustomerId()
 
                 if (dto.EncryptedCustomerId != null)
                 {
-                    _customerId = new Guid(EncryptionUtility.Decrypt(dto.EncryptedCustomerId));
+                    var decryptedCustomerId = new Guid(EncryptionUtility.Decrypt(dto.EncryptedCustomerId));
+
+                    if (dto.IsGuest != true && ValidationPasswordChangedNotDisabled)
+                    {
+                        // GetCustomerByIdAsync uses a cache
+                        var customer = CustomerRepository.GetCustomerByIdAsync(new GetCustomerByIdParam()
+                        {
+                            CustomerId = decryptedCustomerId,
+                            Scope = Scope,
+                            CultureInfo = CultureInfo,
+                            IncludeAddresses =
+                                true // all parameters should be the same as in CustomerViewService.GetAccountHeaderViewModelAsync
```

Comment:
There should be no need to include addresses, as only LastPasswordChanged is used. Or do you need to hit the same cache entry?

Reply:
parameters the same  for cache entry,